packages/flame/test/game/mixins/has_parent_test.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class ParentComponent extends Component {}
+
+class DifferentComponent extends Component {}
+
+class TestComponent extends Component with HasParent<ParentComponent> {}
+
+void main() {
+  group('HasParent', () {
+    testWithFlameGame('successfully sets the parent link', (game) async {
+      final parent = ParentComponent();
+      final component = TestComponent();
+
+      await parent.add(component);
+      await game.add(parent);
+
+      expect(component.parent, isA<ParentComponent>());
+    });
+
+    testWithFlameGame('throws assertion error when the wrong parent is used',
+        (game) async {
+      final parent = DifferentComponent();
+      final component = TestComponent();
+
+      expect(
+        () => parent.add(component),
+        failsAssert('Parent must be of type ParentComponent'),
+      );
+    });
+  });
+}
```

Comment:
@wolfenrain 

Can we add a test that "throws assertion error when changed to the wrong parent"?

Something like:
1. Adds a component to a valid parent
2. Calls `changeParent` to an invalid parent
3. Expect an error

Reply:
as I reworked it to `onMount` for now we can come back to this if we decide to change the way we check it.

Reply:
I think no matter how we implement it, it will be nice to have this tested. 

Personally, adding this test does increase my confidence on the codebase.